### PR TITLE
Symbolic Rate law to Pytorch Rate law

### PR DESCRIPTION
**Pull request by djinnome**
…ook. Will test next.

### CODE REVIEW

**Comment by SamWitty** [comment]
@djinnome , this is tagged as `awaiting review` but there are some failing tests. Is this actually still a WIP?

**Comment by djinnome** [comment]
Hi @SamWitty tests now pass, but I am still blocked on https://github.com/indralab/mira/issues/189
If the AMR `id` is different from the `name`, then there is no easy way to associate the rate law symbols with the appropriate concept.

**Comment by SamWitty** [comment]
Thanks for the update @djinnome . Could you revise the name of this PR to something more informative?

**Review by SamWitty — CHANGES_REQUESTED** [review]
This looks good overall. Could you rerun the integration demos with these changes? This should produce output files without any `mira_param` parameters, and with the solution variable names according to the id, not the long name. Please check this and then I'll approve.

File: src/pyciemss/PetriNetODE/base.py [hunk 1]
```diff
@@ -308,7 +311,41 @@ def __init__(self, G: mira.modeling.Model):
                 getattr(self, f"default_initial_state_{get_name(var)}", None)
                 for var in self.var_order.values()
             )
-
+        setattr(self, "deriv", self.compile_deriv_function())
```
[the remark this answers is not in the document]
Comment (SamWitty) [reply src/pyciemss/PetriNetODE/base.py]: @djinnome could you also address this minor nit?

File: src/pyciemss/PetriNetODE/base.py [hunk 2]
```diff
@@ -308,7 +311,41 @@ def __init__(self, G: mira.modeling.Model):
...
+            for p in t.produced:
+                symbolic_derivs[get_name(p)] += flux
+
+        # convert to a function
+        numeric_derivs = SymPyModule(expressions=list(symbolic_derivs.values()))
+
+        def deriv(self, t: float, state: Tuple[torch.Tensor]) -> Tuple[torch.Tensor]:
+            """Deriv method created on the fly."""
+            # Get the current state
+            states = {v: state[i] for i, v in enumerate(self.var_order.keys())}
```
[the remark this answers is not in the document]
Comment (SamWitty) [reply src/pyciemss/PetriNetODE/base.py]: @djinnome ping?

File: src/pyciemss/PetriNetODE/base.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (djinnome) [reply src/pyciemss/PetriNetODE/base.py]: Line 304 handles this:

```python
if self.compile_rate_law_p:
   self.compiled_rate_law = self.compile_rate_law()
```

File: src/pyciemss/PetriNetODE/base.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (djinnome) [reply src/pyciemss/PetriNetODE/base.py]: `var_order` is created in the `super().__init__()` constructor, but `states` is determined at the time that `deriv` is called, so I can't move this line out of the `deriv` method

**Comment by djinnome** [comment]
I have rerun the `integration_demo` notebook and it now generates output without `aggregate_parameters` by default.

**Review by SamWitty — APPROVED** [review]
Looks great!

**Pull request merged**